Site auth adjustments

roxiciobotea210 opened the pull request:
Here are the adjustments that I made to Sites auth after going through the whole flow for SiteAreas.
In case you agree with everything I did for SiteArea, these changes should also be fine, but I leave it up to you to check and decide. It's mostly refactoring.

jerome-benoit commented on src/server/rest/v1/service/AuthorizationService.ts:
```diff
@@ -49,32 +49,15 @@ export default class AuthorizationService {
 
   public static async checkAndGetSiteAuthorizationFilters(tenant: Tenant, userToken: UserToken,
-      filteredRequest: HttpSiteRequest): Promise<AuthorizationFilter> {
+      filteredRequest: Record<string, any>, authAction: Action): Promise<AuthorizationFilter> {
```
The type change here looks wrong. Why have you make the `filteredRequest` type a lot less strict?

jerome-benoit commented on src/server/rest/v1/service/AuthorizationService.ts:
```diff
@@ -374,7 +323,7 @@ export default class AuthorizationService {
 
   public static async checkAndGetUsersAuthorizationFilters(
-      tenant: Tenant, userToken: UserToken, filteredRequest: HttpUsersRequest): Promise<AuthorizationFilter> {
+      tenant: Tenant, userToken: UserToken, filteredRequest: Record<string, any>): Promise<AuthorizationFilter> {
```
Idem

jerome-benoit commented on src/server/rest/v1/service/AuthorizationService.ts:
```diff
@@ -132,25 +115,9 @@ export default class AuthorizationService {
-      authResult.fields, filteredRequest.ProjectFields);
+
+    await this.canPerformAuthorizationAction(
+      tenant, userToken, Entity.USERS_SITES, Action.LIST, authorizationFilters, filteredRequest);
```
I'm not very fond of the carriage return after the parenthesis code style and Serge neither. 

`await this.canPerformAuthorizationAction(tenant, userToken, Entity.USERS_SITES, Action.LIST, authorizationFilters, filteredRequest)`

roxiciobotea210 commented:
> The code factorisation looks OK but the type weakening is not.
> You can keep the strong typing and use `type1|type2|type3|...` and cast inside methods if needed but the public methods signature must really keep strong types.
> I'm also a bit concerned about the assert removal that I do not see factored out but that's maybe because I've missed something.
> 
> By the way, is that PR still a draft PR? The PR status must be very clear for reviewers, especially at the moment for Serge return.
> 
> Thanks.

I removed the assert because it is the first thing being checked in `AuthorizationService.checkAndGetSiteAuthorization`, which is being called right after the assert statements that I removed.
The type of filteredRequest was weakened so I would be able to use the same method for multiple request types. I will work on changing it as you suggested, thank you for the guidance.

The PR is still in draft state for the moment.

LucasBrazi06 commented:
@roxiciobotea210 Still in draft?

LucasBrazi06 commented:
@roxiciobotea210 Merge conflicts to resolve.
Thanks!

roxiciobotea210 commented:
> @roxiciobotea210 Merge conflicts to resolve.
> Thanks!

I merged the changes but many tests are failing, so please DO NOT MERGE YET!!!

roxiciobotea210 commented:
> @roxiciobotea210 Merge conflicts to resolve.
> Thanks!

I merged the changes but many tests are failing, so please DO NOT MERGE YET!!!

roxiciobotea210 commented:
> > @roxiciobotea210 Merge conflicts to resolve.
> > Thanks!
> 
> I merged the changes but many tests are failing, so please DO NOT MERGE YET!!!

Issue causing tests to fail was fixed. :)

jerome-benoit commented:
> > > @roxiciobotea210 Merge conflicts to resolve.
> > > Thanks!
> > 
> > 
> > I merged the changes but many tests are failing, so please DO NOT MERGE YET!!!
> 
> Issue causing tests to fail was fixed. :)

There's still a some failure on the transaction authorisations.

roxiciobotea210 commented:
> 
> There's still a some failures on the transaction authorisations.

The same tests seem to also be failing on master-qa branch so I'm not so sure the changes in this PR are causing them to fail. But I will look into it.

jerome-benoit commented:
> > There's still a some failures on the transaction authorisations.
> 
> The same tests seem to also be failing on master-qa branch so I'm not so sure the changes in this PR are causing them to fail. But I will look into it.

Ok, that would be great. Could you please then create an issue and a PR when ready linked to it with the fix? 

Thanks.

jerome-benoit commented on src/server/rest/v1/service/AuthorizationService.ts:
```diff
@@ -57,40 +57,24 @@ export default class AuthorizationService {
 
   public static async checkAndGetSiteAuthorizationFilters(tenant: Tenant, userToken: UserToken,
-      filteredRequest: HttpSiteRequest): Promise<AuthorizationFilter> {
+      filteredRequest: Partial<Site>, authAction: Action): Promise<AuthorizationFilter> {
```
Are you sure about that one? It should be `Partial<HttpSiteRequest>`.

roxiciobotea210 replied:
You're right.

jerome-benoit commented on src/server/rest/v1/service/UtilsService.ts:
```diff
@@ -107,7 +106,7 @@ export default class UtilsService {
   public static async checkAndGetChargingStationsAuthorization(tenant: Tenant, userToken: UserToken, action: ServerAction,
       additionalFilters: Record<string, any>, applyProjectFields = false): Promise<ChargingStation[]> {
-    // Check dynamic auth
+  // Check dynamic auth
```
Indentation

jerome-benoit commented on src/server/rest/v1/service/AuthorizationService.ts:
```diff
@@ -313,39 +261,42 @@ export default class AuthorizationService {
 
   public static async checkAndGetUsersAuthorizationFilters(tenant: Tenant, userToken: UserToken,
-      filteredRequest: HttpUsersRequest): Promise<AuthorizationFilter> {
+      filteredRequest?: Partial<HttpUsersRequest>): Promise<AuthorizationFilter> {
```
I think that argument should not be optional, the whole purpose of that method is to check the request.
